Controller/Index/PaymentActionAbstract.php

```diff
@@ -0,0 +1,321 @@
+        JsonFactory $resultJsonFactory,
+        EventManager $eventManager,
+        ConfigHelper $configHelper,
+        Logger $logger
+    ) {
+        parent::__construct($context);
+        $this->resultJsonFactory = $resultJsonFactory;
+        $this->eventManager = $eventManager;
+        $this->configHelper = $configHelper;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param string $eventName
+     * @return $this
+     */
+    public function setEventName($eventName)
+    {
+        $this->eventData[self::EVENT_NAME] = $eventName;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getEventName()
+    {
+        return isset($this->eventData[self::EVENT_NAME]) ? $this->eventData[self::EVENT_NAME] : '';
+    }
+
+    /**
+     * @param string $type
+     * @return string
+     */
+    public function getFullEventName($type = '')
+    {
+        $eventName = $this->getEventName();
+
+        if (strpos($eventName, self::EVENT_NAME_PREFIX) !== false) {
+            $eventName = str_replace(self::EVENT_NAME_PREFIX, '', $eventName);
+        }
+
+        if (strpos($eventName, 'before_') !== false) {
+            $eventName = str_replace('before_', '', $eventName);
+        }
+
+        if (strpos($eventName, 'after_') !== false) {
+            $eventName = str_replace('after_', '', $eventName);
+        }
+
+        if ($type == 'before' || $type == 'after') {
+            $eventName = $type . '_' . $eventName;
+        }
+
+        return self::EVENT_NAME_PREFIX . '_' . $eventName;
+    }
+
+    /**
+     * @param array $eventArgs
+     * @return $this
+     */
+    public function setEventArgs($eventArgs)
+    {
+        $eventArgs = (array)$eventArgs;
+
+        if ([] === $eventArgs) {
+            return $this;
+        }
+
+        if (array_keys($eventArgs) === range(0, count($eventArgs) - 1)) {
+            $eventArgs = array_flip($eventArgs);
+        }
+
+        $this->eventData[self::EVENT_ARGS] = $eventArgs;
+
+        return $this;
+    }
+
+    /**
+     * @return array
+     */
+    public function getEventArgs()
+    {
+        return isset($this->eventData[self::EVENT_ARGS]) ? $this->eventData[self::EVENT_ARGS] : [];
+    }
+
+    /**
+     * @param callable $eventMethod
+     * @return $this
+     */
+    public function setEventMethod($eventMethod)
+    {
+        $this->eventData[self::EVENT_METHOD] = $eventMethod;
+        return $this;
+    }
+
+    /**
+     * @return callable|null
+     */
+    public function getEventMethod()
+    {
+        return isset($this->eventData[self::EVENT_METHOD]) ? $this->eventData[self::EVENT_METHOD] : null;
+    }
+
+    /**
+     * @return string
+     */
+    protected function getEventMethodString()
+    {
+        $eventMethodString = '';
+
+        if (is_callable($this->getEventMethod())) {
+            // phpcs:ignore Magento2.Functions.DiscouragedFunction
+            switch (gettype($this->getEventMethod())) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function gettype() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-magento2-payments/pullRequest?prid=5056328)